tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/aux/ZipOracle.scala

```diff
@@ -0,0 +1,32 @@
+package at.forsyte.apalache.tla.bmcmt.rules.aux
+
+import at.forsyte.apalache.tla.bmcmt.SymbState
+import at.forsyte.apalache.tla.bmcmt.smt.SolverContext
+import at.forsyte.apalache.tla.lir.TypedPredefs.BuilderExAsTyped
+import at.forsyte.apalache.tla.lir.convenience.tla
+import at.forsyte.apalache.tla.lir.{BoolT1, TlaEx}
+
+/**
+ * {{{ZipOracle}}} is an optimization of {{{Oracle}}}. It groups several values of the background oracle together, in
+ * order to reduce the number of constraints. It is up to the user of {{{ZipOracle}}} to make sure that the grouped
+ * values may be treated as equivalent.
+ *
+ * @param backOracle
+ *   the background oracle whose values are grouped together
+ * @param groups
+ *   a list of groups over the values of the background oracle
```

Comment:
It's not clear to me in what sense this is zipping anything. Is it just a general grouping? Or is it establishing pairwise groupings? Having something to clarify what is zippy in the comment would help with readability for me.

Reply:
It zips constraints. Essentially, instead of introduces instances of `k` records, it introduces one instance that can apply to `k` values.

Reply:
I see. Thank you! I do think adding that sentence could help with documentation here, but not essential.

Reply:
I have added the following two sentences:

> In this sense, it compresses several oracle values into one. As a consequence, [[CherryPick]] pick constructs significantly fewer constants and constraints.